app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala

```diff
@@ -205,12 +295,60 @@ class RepresenteeController @Inject() (
             if (answers.fold(_.id, c => Some(c.id)).contains(id))
               Redirect(routes.RepresenteeController.checkYourAnswers())
             else {
+              val name: Option[IndividualName] = answers.fold(i => i.name, c => Some(c.name))
+              val response: EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]] =
+                id match {
+                  case RepresenteeNino(ninoId) =>
+                    businessPartnerRecordService
+                      .getBusinessPartnerRecord(IndividualBusinessPartnerRecordRequest(Right(ninoId), name))
+                      .map(bpr => Some(Left(bpr)))
+                  case RepresenteeSautr(sautr) =>
+                    businessPartnerRecordService
+                      .getBusinessPartnerRecord(IndividualBusinessPartnerRecordRequest(Left(sautr), name))
+                      .map(bpr => Some(Left(bpr)))
+                  case RepresenteeCgtReference(cgtReference) => {
+                    subscriptionService.getSubscribedDetails(cgtReference).map(sd => Some(Right(sd)))
```

Comment:
you could do the name matching here for cgt ref's - that would mean you wouldn't need `nameCheck` below which saves some lines

Reply:
Yes good point. Done.